WordPressKit/StatsServiceRemoteV2.swift

```diff
@@ -0,0 +1,193 @@
+        })
+    }
+
+    // "Last Post" Insights are "fun" in the way that they require multiple requests to actually create them,
+    // so we do this "fun" dance in a separate method.
+    public func getInsight(completion: @escaping ((StatsLastPostInsight?, Error?) -> Void)) {
+         getLastPostInsight(completion: completion)
+    }
+
+    private func getLastPostInsight(completion: @escaping ((StatsLastPostInsight?, Error?) -> Void)) {
+        let properties = StatsLastPostInsight.queryProperties
+        let pathComponent = StatsLastPostInsight.pathComponent
+
+        let path = self.path(forEndpoint: "sites/\(siteID)/\(pathComponent)", withVersion: ._1_1)
+
+        wordPressComRestApi.GET(path, parameters: properties, success: { (response, _) in
+            guard
+                let jsonResponse = response as? [String: AnyObject],
+                let posts = jsonResponse["posts"] as? [[String: AnyObject]],
+                let post = posts.first,
+                let postID = post["ID"] as? Int else {
+                    completion(nil, ResponseError.decodingFailure)
+                    return
+            }
+
+            self.getPostViews(for: postID) { (views, error) in
+                guard
+                    let views = views,
+                    let insight = StatsLastPostInsight(jsonDictionary: post, views: views) else {
+                        completion(nil, ResponseError.decodingFailure)
+                        return
+
+                }
+
+                completion(insight, nil)
+            }
+        }, failure: {(error, _) in
+            completion(nil, error)
+        })
+    }
+
+    public func getPostViews(`for` postID: Int, completion: @escaping ((Int?, Error?) -> Void)) {
+        let parameters = ["fields": "views" as AnyObject]
+
+        let path = self.path(forEndpoint: "sites/\(siteID)/stats/post/\(postID)", withVersion: ._1_1)
+
+        wordPressComRestApi.GET(path,
+                                parameters: parameters,
+                                success: { (response, _) in
+                                    guard
+                                        let jsonResponse = response as? [String: AnyObject],
+                                        let views = jsonResponse["views"] as? Int else {
+                                            completion(nil, ResponseError.decodingFailure)
+                                            return
+                                    }
+                                    completion(views, nil)
+                                }, failure:  { (error, _) in
+                                    completion(nil, error)
+                                }
+        )
+    }
+
+}
+
+// This serves both as a way to get the query properties in a "nice" way,
+// but also as a way to narrow down the generic type in `getInisght:` method.
+public protocol InsightProtocol {
+    static var queryProperties: [String: AnyObject] { get }
```

Comment:
A fair number of these implementations append `as AnyObject` to conform to this type.

I wonder if it be worthwhile for this type to be `[String: String?]` instead. In my mind, this would offer the following benefits:

- It matches the semantics of [`URLQueryItem`](https://developer.apple.com/documentation/foundation/urlqueryitem), which would benefit us if we ever consider `URLSession` over `Alamofire`.
- It should be easily cast to `[String: AnyObject]` in/on `StatsServiceRemoteV2.swift:24`

Thoughts?

Reply:
Good point!